src/app/zap-templates/templates/chip/helper.js

```diff
@@ -271,6 +271,19 @@ function isManufacturerSpecificCommand()
   return !!this.mfgCode;
 }
 
+function valid_callback_attribute_types(options)
+{
+  // Valid types should match what can be returned in asCallbackAttributeType() below.
+  const validTypesList =
+      [ 'List', 'String', 'Boolean', 'Int8u', 'Int16u', 'Int32u', 'Int64u', 'Int8s', 'Int16s', 'Int32s', 'Int64s' ];
+  const validTypes = validTypesList.map(type => {
+    return {
+      'type': type
+    }
+  });
+  return asBlocks.call(this, Promise.resolve(validTypes), options);
+}
```

Comment:
It makes sense to have an iterator for the base callbacks types.

That would be useful for the code gen of other platforms as well.

But instead of making it that way and add more types conversions, Can you add an iterator that consumes all the atomics and create a list of "callbacks" from them ?

That would replace`valid_callback_attribute_types` and `asCallbackAttributeBasicType` by a new iterator that  carries both the original {{type}} and the {{chipType}}. That should allow to remove `attributeCallbackTypeStringAsCType`  and prevent custom checks for types such as `{{#if (isStrEqual type "String")}}` instead of the helpers from `StringHelper.js`